File: src/ltd/rust_lang/utils/queue/Queue.java
package ltd.rust_lang.utils.queue;

import ltd.rust_lang.utils.list.DoublyLinkList;
import ltd.rust_lang.utils.list.MyList;

public class Queue<E> {
  private MyList<E> list = new DoublyLinkList<>();

  public int size() {
    return list.size();
  }

  public boolean isEmpty() {
    return list.isEmpty();
  }

  public void enQueue(E element) {
    list.add(element);
  }

  public E deQueue() {
    return list.remove(0);
  }

  public E front() {
    return list.get(0);
  }

  public void clear() {
    list.clear();
  }
}

File: src/main/kotlin/ltd/rust_lang/utils/queue/Queue.kt
package ltd.rust_lang.utils.queue

import ltd.rust_lang.utils.list.LinkedList

class Queue<E> {
    private val list = LinkedList<E>()

    fun peek(): E? {
        return list[0]
    }

    fun size(): Int {
        return list.size()
    }

    val isEmpty: Boolean
        get() = list.isEmpty

    fun offer(e: E?): Boolean {
        return list.add(e)
    }

    fun poll(): E? {
        return list.remove(0)
    }

    fun clear() {
        list.clear()
    }
}
